modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -127,15 +128,160 @@ package object widgets {
 
 
   def html(html: NodeSeq): Widget = new SimpleWidget(html)
+  
+  def barChart(width: Int, jsons: Seq[(String, Any)]) = {
+    val id = Math.abs(Random.nextInt).toString
+    <div>
+    	<svg id={ "bar"+id } width="600px" height="400px"
+       xmlns="http://www.w3.org/2000/svg" version="1.1">
+    	{
+      scopedScript(
+        s""" req(
+              ['d3', 'dimple'],
+              function (O, x) {
+                var svg = d3.select("#bar${id}");
+    		  	var chart = new dimple.chart(svg, data);
+        		chart.setBounds(50, 20, 540, 350);
+    		  	chart.addCategoryAxis("x", "${jsons(0)._1.trim}");
+    		  	chart.addMeasureAxis("y", "${jsons(1)._1.trim}");
+    		  	chart.addSeries(null, dimple.plot.bar);
+    		  	chart.draw();
+          
+              });
+        """,
+        Json.obj( "data" -> (jsons grouped width map { row =>  Json.obj(row.map( f => f._1.trim -> toJson(f._2) ):_*)  }).toSeq    )
+      )
+    } </svg></div>
+    
+  }
+  
+  def lineChart(width: Int, jsons: Seq[(String, Any)]) = {
+    val id = Math.abs(Random.nextInt).toString
+    <div>
+    	<svg id={ "line"+id } width="600px" height="400px"
+       xmlns="http://www.w3.org/2000/svg" version="1.1">
+    	{
+      scopedScript(
```

Comment:
We can try to externalize this,  but if it's work I'm more than happy.
If possible using the injected deps would be better, but I know it's always tricky is Js
I'll take a look my self when merged ;-)